grails-app/views/taglib/_decimalFormatField.gsp

```diff
@@ -16,7 +16,18 @@
         .on("input", function () {
           const sign = this.value.substring(0, 1) === "-" ? "-" : ""
           // prevent user from inputting any other characters than: digits or decimal separator of the current locale
-          const parsedNumber = this.value.replace(/(?![\d"${decimalSeparator}"])./g, "");
+          let parsedNumber = this.value.replace(/(?![\d"${decimalSeparator}"])./g, "");
+
+          // handle decimal points
+          if ("${attrs.decimal}") {
+            const decimalPoint = parseInt("${attrs.decimal}")
+            const [number, decimal] = parsedNumber?.split("${decimalSeparator}");
+            if (decimalPoint > 0) {
+              parsedNumber = number + "${decimalSeparator}" + decimal.substring(0, decimalPoint);
+            } else {
+              parsedNumber = number;
+            }
+          }
```

Comment:
I don't understand this part. Could you elaborate on that? The logic seems a bit complexed and is difficult to understand without proper examples?

Reply:
I was afraid that this might not be so straightforward at first glance but I tried making this logic as simple as I could.
I guess adding comment to each line would explain this process
```js
// only if attribute decimal is passed then apply below logic
if ("${attrs.decimal}") {
  // attrs.decimal is a string so we need to cast it to a number
  const decimalPoint = parseInt("${attrs.decimal}")
  // given we have a number "12,345"  with a decimal separator ","
  // we split the number string into two parts: number (12) and decimal (345)
  const [number, decimal] = parsedNumber?.split("${decimalSeparator}");
  if (decimalPoint > 0) {
    // if decimalPoint is > 0 (for example 2), then return  a number with fixed decimal points like: "12" + "," + "34" 
    parsedNumber = number + "${decimalSeparator}" + decimal.substring(0, decimalPoint);
  } else {
  // if decimal point is <= 0 the just return the number and ignore the decimal part
  parsedNumber = number;
}
```

Reply:
this is what was missing :laughing: 
Thanks, I've had a good grasp of what is going on, but wanted you to confirm